[BUG] Property resolution leads to SEGV when POM has no own props

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

How can I get it? I am using Analyzer-LSP directly!

### Priority

Blocker

### Current Behavior

When analysing a Java POM with a dependency as a property, analyzer tries to resolve the property in the same POM. If the POM contains no properties at all, this leads to a segmentation violation (in [line 141 of `provider/internal/java/dependeny.go`](https://github.com/konveyor/analyzer-lsp/blob/9369083d47c8b056775300b38adeae856612f397/provider/internal/java/dependency.go#L141)):

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5b3ddc]
```


### Expected Behavior

Try to resolve property via parent POM (hierarchy) or leave unresolved otherwise (which [is the default, if the property cannot be found in an existing POM property list](https://github.com/konveyor/analyzer-lsp/blob/9369083d47c8b056775300b38adeae856612f397/provider/internal/java/dependency.go#L142)).

### How Reproducible

Always (Default)

### Steps To Reproduce

Prepare a project with POM like this.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>some-project</groupId>
    <artifactId>some-artifact</artifactId>
    <version>does-not-matter</version>
    
    <dependencies>
        <dependency>
            <groupId>some-dep-group</groupId>
            <artifactId>some-dep-artifact</artifactId>
            <version>${cannot-be-resolved}</version>
        </dependency>
    </dependencies>
</project>
``` 

And analyze it.

### Environment

```markdown
Have to provide a minimal example ...
```


### Anything else?

_No response_

--- a/provider/internal/java/dependency.go
+++ b/provider/internal/java/dependency.go
@@ -111,6 +111,9 @@ func (p *javaServiceClient) GetDependenciesFallback(ctx context.Context, locatio
 		path = location
 	}
 	pom, err := gopom.Parse(path)
+	p.log.V(10).Info("Analyzing POM",
+		"POM", fmt.Sprintf("%s:%s:%s", pomCoordinate(pom.GroupID), pomCoordinate(pom.ArtifactID), pomCoordinate(pom.Version)),
+		"error", err)
 	if err != nil {
 		return nil, err
 	}
@@ -140,9 +143,20 @@ func (p *javaServiceClient) GetDependenciesFallback(ctx context.Context, locatio
 		if *d.Version != "" {
 			if strings.Contains(*d.Version, "$") {
 				version := strings.TrimSuffix(strings.TrimPrefix(*d.Version, "${"), "}")
-				version = pom.Properties.Entries[version]
-				if version != "" {
+				p.log.V(10).Info("Searching for property in properties",
+					"property", version,
+					"properties", pom.Properties)
+				if pom.Properties == nil {
+					p.log.Info("Cannot resolve version property value as POM does not have properties",
+						"POM", fmt.Sprintf("%s.%s", pomCoordinate(pom.GroupID), pomCoordinate(pom.ArtifactID)),
+						"property", version,
+						"dependency", dep.Name)
 					dep.Version = version
+				} else {
+					version = pom.Properties.Entries[version]
+					if version != "" {
+						dep.Version = version
+					}
 				}
 			} else {
 				dep.Version = *d.Version
@@ -176,6 +190,13 @@ func (p *javaServiceClient) GetDependenciesFallback(ctx context.Context, locatio
 	}
 
 	return m, nil
+}
+
+func pomCoordinate(value *string) string {
+	if value != nil {
+		return *value
+	}
+	return "unknown"
 }
 
 func (p *javaServiceClient) GetDependenciesDAG(ctx context.Context) (map[uri.URI][]provider.DepDAGItem, error) {

